tools/document-pii/src/main.rs

```diff
@@ -0,0 +1,696 @@
+        UseTree::Name(use_name) => {
+            leading_path.segments.push(use_name.ident.clone().into());
+            quote::quote!(#leading_path).to_string()
+        }
+        UseTree::Group(use_group) => {
+            let mut paths = Vec::new();
+            for item in &use_group.items {
+                paths.push(use_tree_to_path(leading_path.clone(), item));
+            }
+            paths.join(", ")
+        }
+
+        UseTree::Rename(use_rename) => {
+            leading_path.segments.push(use_rename.rename.clone().into());
+            quote::quote!(#leading_path).to_string()
+        }
+        _ => quote::quote!(#leading_path).to_string(),
+    }
+}
+
+fn print_error(error: &anyhow::Error) {
+    eprintln!("Error: {error}");
+
+    let mut cause = error.source();
+    while let Some(ref e) = cause {
+        eprintln!("  caused by: {e}");
+        cause = e.source();
+    }
+}
+
+fn main() {
+    let cli = Cli::parse();
+
+    match cli.run() {
+        Ok(()) => (),
+        Err(error) => {
+            print_error(&error);
+            std::process::exit(1);
+        }
+    }
+}
+
+/// Prints documentation for metrics.
+#[derive(Debug, Parser, Default)]
+#[command(verbatim_doc_comment)]
+pub struct Cli {
+    /// The format to output the documentation in.
+    #[arg(value_enum, short, long, default_value = "json")]
+    pub format: SchemaFormat,
+
+    /// Optional output path. By default, documentation is printed on stdout.
+    #[arg(short, long)]
+    pub output: Option<PathBuf>,
+
+    /// Path to the rust crate/workspace
+    pub path: Option<PathBuf>,
+
+    /// The struct or enum of which you want to find all pii fields. Checks all items if none is
+    /// provided.
+    #[arg(short, long)]
+    pub item: Option<String>,
+
+    /// Vector of which pii-values should be looked for, options are: "true, maybe, false".
+    #[arg(long, default_value = "true")]
+    pub pii_values: Vec<String>,
+}
+
+impl Cli {
+    fn write_pii<W: Write>(&self, writer: W, metrics: &[Pii]) -> anyhow::Result<()> {
+        match self.format {
+            SchemaFormat::Json => serde_json::to_writer_pretty(writer, metrics)?,
+            SchemaFormat::Yaml => serde_yaml::to_writer(writer, metrics)?,
+        };
+
+        Ok(())
+    }
+
+    pub fn run(self) -> anyhow::Result<()> {
+        let rust_file_paths = {
+            let path = self
+                .path
+                .clone()
+                .unwrap_or_else(|| std::env::current_dir().unwrap());
+
+            find_rs_files(&path)
+        };
+
+        // Before we can iterate over the pii fields properly, we make a mapping between all
+        // paths to types and their AST node.
+        FileSyntaxVisitor::fill_pii_hashmaps(&rust_file_paths);
+
+        {
+            // Sets the pii_values from Cli to a global variable so it can be accessed in the
+            // 'Visit' trait.
+            let mut guard = PII_VALUES.lock().unwrap();
+            guard.clear();
```

Comment:
At the start of the program the global variables are empty. Is there any reason why we are clearing an already-clear vector?